src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -26,7 +26,7 @@ public static void WriteRequestCreation(CodeWriter writer, RestClientMethod clie
 
             var methodName = CreateRequestMethodName(clientMethod.Name);
             var returnType = lowLevel ? typeof(Azure.Core.Request) : typeof(HttpMessage);
-            var visibility = lowLevel ? "private" : (clientMethod.IsVisible ? "protected" : "internal");
+            var visibility = clientMethod.Accessibility ?? (lowLevel ? "private" : "internal");
```

Comment:
I think we should move the default calculation to the modeling layer. How do you think it would be?

Reply:
That's fair (and I don't think that difficult).